src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -512,26 +509,52 @@ TypeSymbol BindRecursivePatternType(TypeSyntax typeSyntax, TypeSymbol inputType,
             }
         }
 
-        private BoundPattern BindDeconstructionPattern(DeconstructionPatternSyntax node, TypeSymbol inputType, bool hasErrors, DiagnosticBag diagnostics)
+        private BoundPattern BindRecursivePattern(RecursivePatternSyntax node, TypeSymbol inputType, bool hasErrors, DiagnosticBag diagnostics)
         {
             TypeSyntax typeSyntax = node.Type;
             TypeSymbol declType = BindRecursivePatternType(typeSyntax, inputType, diagnostics, ref hasErrors, out BoundTypeExpression boundDeclType);
 
-            var patterns = ArrayBuilder<BoundSubpattern>.GetInstance(node.SubPatterns.Count);
             MethodSymbol deconstructMethod = null;
+            ImmutableArray<BoundSubpattern> deconstructionSubpatterns = default;
+            if (node.DeconstructionPatternClause != null)
+            {
+                deconstructionSubpatterns = BindDeconstructionPatternClause(node.DeconstructionPatternClause, declType, diagnostics, out deconstructMethod, ref hasErrors);
+            }
+
+            ImmutableArray<BoundSubpattern> properties = default;
+            if (node.PropertyPatternClause != null)
+            {
+                properties = BindPropertyPatternClause(node.PropertyPatternClause, declType, diagnostics, ref hasErrors);
+            }
+
```

Comment:
It can be null here and `BindPatternDesignation` handles that the same as a discard designation.

---
In reply to: [189670479](https://github.com/dotnet/roslyn/pull/26782#discussion_r189670479) [](ancestors = 189670479)